lib/till.rb

```diff
@@ -0,0 +1,87 @@
+class Till
+
+  def self.coffee_shop=(shop)
+    @coffee_shop = shop
+  end
+
+  def self.customer_order=(order)
+    @customer_order = order
+  end
+
+  def self.line_entries
+    prices = coffee_shop[:prices]
+    customer_order[:order].map { |item, quantity|
+      [item, {
+        quantity: quantity,
+        item_price: prices[item],
+        line_total: quantity * prices[item]
+      }]
+    }.to_h
+  end
+
+  def self.total_before_discounts
+    line_entries.to_a.map { |item| item[1][:line_total] }.inject(&:+).round(2)
+  end
+
+  def self.discounts
+    discounts = {}
+    prices = coffee_shop[:prices]
+    item_discounts = coffee_shop[:itemDiscounts]
+    bill_discount = coffee_shop[:billDiscount]
+    threshold = bill_discount[:threshold]
+    customer_order[:order].each do |item, quantity|
+      rate = item_discounts[item]
+      if rate
```

Comment:
Style/Next: Use `next` to skip iteration. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)